staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -314,3 +317,41 @@ func (e *eventBroadcasterImpl) StartRecordingToSink(stopCh <-chan struct{}) {
 		stopWatcher()
 	}()
 }
+
+type eventClientImpl struct {
+	CoreClient         typedv1core.EventsGetter
+	CoreBroadcaster    record.EventBroadcaster
+	V1beta1Client      typedv1beta1.EventsGetter
+	V1beta1Broadcaster EventBroadcaster
+}
+
+// NewEventClient Creates a new event broadcaster.
+func NewEventClient(client clientset.Interface) *EventClient {
+	eventClient := &eventClientImpl{}
+	if _, err := client.Discovery().ServerResourcesForGroupVersion(v1beta1.SchemeGroupVersion.String()); err == nil {
+		eventClient.V1beta1Client = client.EventsV1beta1()
+		eventClient.V1beta1Broadcaster = NewBroadcaster(&EventSinkImpl{Interface: eventClient.V1beta1Client.Events("")})
+	} else {
+		eventClient.CoreClient = client.CoreV1()
+		eventClient.CoreBroadcaster = record.NewBroadcaster()
+	}
+	return eventClient
+}
+
+// StartRecordingToSink starts sending events received from the specified eventBroadcaster to the given sink.
+func (e *eventClientImpl) StartRecordingToSink(stopCh <-chan struct{}) {
+	if e.V1beta1Broadcaster != nil && e.V1beta1Client != nil {
+		e.V1beta1Broadcaster.StartRecordingToSink(stopCh)
+	} else {
```

Comment:
you can remove the else here

Reply:
Why - I think you need it...

Reply:
ha you're right, forgot that `StartRecordingToSink` didn't return something